### DISCUSSION TITLE
Questions Regarding Current Support and Stability for Spark 4.0

### DISCUSSION THREAD

**Discussion by Echan0808** [apache/kyuubi#6630] [Q&A] [answered]
I noticed in the latest release notes that there is "experimental support for Spark 4.0" mentioned, along with some feature improvements like the TPC-H/DS connector and SPARK_HOME detection that now support Spark 4.0. I'm considering using Spark 4.0 in our project and have a few questions:

1. What does the experimental support for Spark 4.0 specifically cover?
  - Which features have been thoroughly tested, and which might still have compatibility issues or are not fully adapted to Spark 4.0?

2. Is it safe to use Spark 4.0 in a production environment?
  - Are there any known limitations or potential issues to be aware of? Has anyone in the community successfully run Spark 4.0 in production?

3. What are the future plans for Spark 4.0 support?
  - Is there a timeline for releasing a stable version with full Spark 4.0 support? What further improvements and testing are planned?

Thank you～

**Comment by pan3793** [comment, marked as answer]
@Echan0808 thanks for interested in the Spark 4.0 support.

> 1. What does the experimental support for Spark 4.0 specifically cover?

basically in two aspects. firstly, the current integration tests are based on the Spark [4.0.0-preview1](https://spark.apache.org/news/spark-4.0.0-preview1.html), which is not a stable release thus does not guarantee stable API, that means it may not fully compatible with the final Spark 4.0.0. secondly, we have made all functionalities in the Spark SQL engine work with Spark 4.0.0-preview1, which means you can use the Kyuubi to bootstrap Spark 4.0.0-preview1 applications, run SQL/Python/Scala quires and retrieve results, etc. but plugins like authz, spark extensions do not work yet.

> 2. Is it safe to use Spark 4.0 in a production environment?

Please read https://spark.apache.org/news/spark-4.0.0-preview1.html.

> 3. What are the future plans for Spark 4.0 support?

Kyuubi will publish a version soon(usually in a few days) after Spark 4.0.0 is out, with full Spark 4.0 supported. AFAIK, the Spark 4.0.0-preview2 is coming soon, and the final 4.0.0 is likely out in Q4. You can [subscribe Spark mailing list](https://spark.apache.org/community.html) to receive more information

**Reply by Echan0808** [reply]
Thank you very much. Your insights are incredibly helpful and greatly appreciated~
